test/plain-list/AutosuggestApp.test.js

```diff
@@ -131,6 +132,102 @@ describe('Plain list Autosuggest', () => {
     });
   });
 
+  describe('when typing and matches exist (when selectFirstSuggestion is true)', () => {
+    before(() => {
+      setSelectFirstSuggestion(true);
+    });
+
+    after(() => {
+      setSelectFirstSuggestion(false);
+    });
+
+    beforeEach(() => {
+      onSuggestionSelected.reset();
+      focusAndSetInputValue('p');
+    });
+
+    it('should focus on any suggestion', () => {
+      expectFocusedSuggestion('Perl');
+    });
+
+    it('should hide suggestions when Escape is pressed', () => {
+      clickEscape();
+      expectSuggestions([]);
+    });
```

Comment:
This is one example of a repeated test (same test for `selectFirstSuggestion = true` and `selectFirstSuggestion = false`). Although I can see the value in this repetition, I'm not sure it worth the overhead it adds to the maintenance of the test suite. Just by adding this `selectFirstSuggestion` boolean prop to Autosuggest, this test file grew from 665 to 1025 lines! I suggest we add only tests that are specific to `selectFirstSuggestion = true`. Thoughts?